### ISSUE TITLE
feat(r2dreamer/encoder): add nonlinearity to VGGTEncoder for pose×geometry interactions

### ISSUE THREAD

**Issue by LSailer** [LSailer/Master-Thesis-3D-VLA#89]
## Problem

`VGGTEncoder` is a single `Dense(4116 → 1024)` layer with no nonlinearity (networks.py:414–425). This means the encoder can only compute **additive** combinations of input features — it cannot model **multiplicative interactions**.

For VGGT inputs specifically, this is a real limitation: the same `world_points` mean different things depending on `camera_pose` orientation. Encoding \"obstacle is on my left\" requires representing \"wall_in_front × pose_facing_left\" — a multiplicative gate that a linear projection cannot express.

## Why this matters for the thesis

A linear encoder reduces VGGT's rich 3D output to a learned linear subspace (essentially PCA-like). It preserves variance and decorrelates dimensions, but cannot:
- Model spatial-temporal feature interactions
- Highlight a small but contextually important signal (e.g., pose) via gating
- Represent \"if X then Y\" relationships between geometry and pose

This is a confounder for the 3D-vs-2D experiment: if 3D ties or loses, we cannot distinguish \"3D doesn't help\" from \"a more expressive encoder would have extracted what we missed.\"

## Relationship to #87

This is **orthogonal** to #87 (pose vs geometry rebalancing):

| Issue | Layer | Fix |
|-------|-------|-----|
| #87   | Input-side (signal share) | LayerNorm per modality, or two separate projections |
| This  | Architecture-side (expressivity) | Nonlinearity + extra layer |

- Fix only #87 → pose has signal, but no interactions modeled
- Fix only this → interactions possible, but pose still drowned in geometry
- Fix both → encoder can both *see* and *combine* pose

## Why it's a known DreamerV3 anti-pattern

Official DreamerV3 uses a CNN encoder (nonlinear). The R2-Dreamer port likely chose a single Dense as the simplest drop-in for already-preprocessed VGGT features. Reasonable starting point, but worth replacing for production runs.

The downstream RSSM is nonlinear (GRU with tanh/sigmoid) and could *theoretically* recover interactions — but only if the relevant information survives the linear bottleneck. Information lost at the encoder cannot be recreated downstream.

## Proposed fix

Replace single Dense with a small MLP. Minimal change:

```python
class VGGTEncoder(nn.Module):
    embed_dim: int = 1024
    hidden_dim: int = 1024
    @nn.compact
    def __call__(self, obs):
        x = nn.Dense(self.hidden_dim, name=\"hidden\")(obs)
        x = nn.gelu(x)
        x = nn.Dense(self.embed_dim, name=\"proj\")(x)
        return x
```

GELU chosen over ReLU because: (a) DreamerV3 uses it, (b) smoother gradients near zero, (c) better with normalization layers.

## Acceptance criteria

- [ ] Implement MLP encoder in `modules/r2dreamer/networks.py::VGGTEncoder`
- [ ] Add a parametrized test in `modules/r2dreamer/tests/test_vggt_encoder.py` that confirms the encoder can express XOR-like patterns (an explicit proof of nonlinearity)
- [ ] Document the design decision in `docs/wiki/methods/` (cite this issue + the linear-encoder limitation)
- [ ] Run an A/B comparison (linear vs MLP encoder) on a short ObjectNav training and report whether SR / sample efficiency changes

## Origin

Surfaced during interactive debug walkthrough (BP #4). Related to #87 (pose dilution).
